contents/rust/Rust.md

```diff
@@ -0,0 +1,80 @@
+
+Concurrency is getting [increasingly important](https://softwareengineering.stackexchange.com/questions/115474/why-should-i-know-concurrent-programming) but it is challenging to write concurrent code ([[1]](https://news.ycombinator.com/item?id=8138578), [[2]](https://golang.org/doc/faq#csp), [[3]](http://joeduffyblog.com/2016/11/30/15-years-of-concurrency/)).
+
+Rust provides concurrency which is built upon the safety concepts. The implication is that the safety concepts allows us to be fearless when writing concurrent code by helping pointing out mistakes compile time [[1]](https://blog.rust-lang.org/2015/04/10/Fearless-Concurrency.html).
+
+### Practical
+
+Rust is designed to be practical, but this term is vague and does not say much. Instead, we can take a look at some points of Rust's guiding design:
+* Uses old established techniques instead of particularly cutting-edge technologies
+* Provides only majority-case features
+* Runs on widely used platforms without unnecessary compromises
+
+See [non-goals](https://www.rust-lang.org/en-US/faq.html#what-are-some-non-goals).
+
+#### Production Ready
+Also, in order for a language to be practical it must be usable in production. Aside from [Servo](https://github.com/servo/servo), the prototype browser engine Mozilla is working on, Rust is also used in companies such as Dropbox [[1]](https://news.ycombinator.com/item?id=11283688). 
+
+#### Developer experience
+Practicality can also be measured with how much developers enjoy using the language and want to continue using it, because if the language is not very practical the developer experience would not be very good. For Rust, it is a language that developers want to continue using ([[1]](https://www.reddit.com/r/rust/comments/842adc/rust_voted_most_loved_language_for_the_3rd_year/dvmftnk/), [[2]](https://medium.com/mozilla-tech/why-rust-is-the-most-loved-language-by-developers-666add782563)).
+
+Although Rust does not use particularly cutting-edge technologies, Rust is still modern in that it is [significantly influenced by functional programming](https://doc.rust-lang.org/book/second-edition/ch13-00-functional-features.html), and has a type system that is [drawn from Haskell's typeclasses](https://www.rust-lang.org/en-US/faq.html#compare-go-and-rust). This allows Rust to also benefit from the advantages of functional programming. 
+
+There are also a lot of other details that go into making a great developer experience, such as tooling and documentation. However, instead of going into more details, let's look at a resource that summarizes why others think Rust is great:
+
+[fireflowers - The Rust Programming Language, in the words of its practitioners](https://brson.github.io/fireflowers/)
+
+## How to learn Rust
+
+Knowing the merits of a language without knowing how to learn it however is not sufficient. As mentioned, part of making the developer experience great is good documentation. And Rust has many great resources to learn it.
```

Comment:
`Knowing the merits of a language without knowing how to learn it however is not sufficient. As mentioned, part of making the developer experience great is good documentation. And Rust has many great resources to learn it.` -> `Knowing the merits of a language without knowing how to learn it is not sufficient. As [mentioned](#developer-experience), part of making the developer experience great is good documentation. And there are many great avenues to learn Rust.`